agent/structs/structs.go

```diff
@@ -2391,3 +2410,18 @@ func (r *KeyringResponses) Add(v interface{}) {
 func (r *KeyringResponses) New() interface{} {
 	return new(KeyringResponses)
 }
+
+type UpstreamDownstream struct {
+	Upstream   ServiceName
+	Downstream ServiceName
+
+	// Upstream/Downstream pairs come from individual service registrations, and they can be updated individually.
```

Comment:
Did you mean for this comment to be associated with the Upstream/Downstream fields?